pkg/providers/tinkerbell/assert.go

```diff
@@ -313,6 +334,26 @@ func (v *ValidatableTinkerbellCAPI) WorkerNodeHardwareGroups() []WorkerNodeHardw
 	return workerNodeHardwareList
 }
 
+// ClusterK8sVersion returns the Kubernetes version in major.minor format for a ValidatableTinkerbellCAPI.
+func (v *ValidatableTinkerbellCAPI) ClusterK8sVersion() *v1alpha1.KubernetesVersion {
+	return v.toK8sVersion(v.KubeadmControlPlane.Spec.Version)
+}
+
+// WorkerNodeGroupK8sVersion returns each worker node group mapped to Kubernetes version in major.minor format for a ValidatableTinkerbellCAPI.
+func (v *ValidatableTinkerbellCAPI) WorkerNodeGroupK8sVersion() map[string]*v1alpha1.KubernetesVersion {
+	wngK8sversion := make(map[string]*v1alpha1.KubernetesVersion)
+	for _, wng := range v.WorkerGroups {
+		k8sVersion := v.toK8sVersion(*wng.MachineDeployment.Spec.Template.Spec.Version)
+		wngK8sversion[wng.MachineDeployment.Name] = k8sVersion
+	}
+	return wngK8sversion
+}
+
+func (v *ValidatableTinkerbellCAPI) toK8sVersion(k8sversion string) *v1alpha1.KubernetesVersion {
```

Comment:
Why are we using pointers for `KubernetesVersion`? From what I could tell all consumers dereference it anyway?